components/automate-cli/cmd/chef-automate/cert.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/spf13/cobra"
+
+	"github.com/chef/automate/api/config/shared"
+	"github.com/chef/automate/components/automate-cli/pkg/status"
+	"github.com/chef/automate/components/automate-deployment/pkg/client"
+)
+
+type certCmdFlagSet struct {
+	hostname string
+	filepath string
+}
+
+var certCmdFlags = certCmdFlagSet{}
+
+func init() {
+
+	certCmd := certCmd()
+
+	certCmd.PersistentFlags().StringVarP(
+		&certCmdFlags.hostname,
+		"hostname",
+		"n",
+		"",
+		"Hostname for the automate TLS certificate",
+	)
+	certCmd.PersistentFlags().StringVarP(
+		&certCmdFlags.filepath,
+		"file",
+		"f",
+		"",
+		"File path to save automate TLS certifcate to.",
+	)
+	RootCmd.AddCommand(certCmd)
+}
+
+func certCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "external-cert COMMAND",
+		Short: "Manage Chef Automate's external certificate",
+		Long:  "Manage Chef Automate's external certificate authority. Used for establishing TLS/SSL communication with automate.",
+	}
+
+	certShow := &cobra.Command{
+		Use:   "show",
+		Short: "Show the external TLS/SSL certificates in Automate. Optionally, save the certificates to a file in the specified path.",
+		RunE:  runCertShowCmd,
+		Args:  cobra.MaximumNArgs(2),
+	}
+
+	cmd.AddCommand(certShow)
+
+	return cmd
+}
+
+func runCertShowCmd(*cobra.Command, []string) error {
+	res, err := client.GetAutomateConfig(int64(client.DefaultClientTimeout))
+	if err != nil {
+		return status.Wrap(
+			err,
+			status.DeploymentServiceUnreachableError,
+			"Connecting to deployment-service failed",
+		)
+	}
+	tlsCreds := res.Config.GetGlobal().GetV1().GetFrontendTls()
+
+	if certCmdFlags.hostname != "" && certCmdFlags.filepath != "" {
+		for _, tlsCred := range tlsCreds {
+			if tlsCred.GetServerName() == certCmdFlags.hostname {
+				return writeToFile(tlsCred.GetCert())
+			}
+		}
+
+		hostnames := gatherHostnames(tlsCreds)
+		// Error no matching hostname found in the automate frontend_tls cert configuration
+		return status.Wrap(
+			err,
+			status.CommandExecutionError,
+			fmt.Sprintf("Unable to find matching certificate configuration for hostname %s. Available Hostnames: %q", certCmdFlags.filepath, hostnames),
+		)
```

Comment:
This error message is a little odd when the user has something like our current dev config:

```
CommandExecutionError: An issue occurred when running an executable command: Unable to find matching certificate configuration for hostname . Available hostnames: [""]
```

We don't require `server_name` to be set in the config (we will fill it in with `global.v1.fqdn`) if it is unset.

Reply:
It's a weird case, because if no hostname is set then we don't want the user using the hostname flag. Maybe an error message saying no hostname is set? Then that is more logic to check it.